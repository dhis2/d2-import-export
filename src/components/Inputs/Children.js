import React from 'react'
import i18n from '@dhis2/d2-i18n'

import { Field } from '../Field/Field'
import { Label } from '../Field/Label'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_YES = {
    value: 'true',
    label: i18n.t('Include descendants of organisation unit'),
}
export const OPTION_NO = {
    value: 'false',
    label: i18n.t('Selected organisation unit'),
}
export const CHILDREN_KEY = 'children'
export const CHILDREN_DEFAULT_VALUE = OPTION_YES.value

export const Children = () => (
    <Field>
        <Label>{i18n.t('Children')}</Label>
        <RadioGroup
            name={CHILDREN_KEY}
            label={i18n.t('Children')}
            options={[OPTION_YES, OPTION_NO]}
        />
    </Field>
)
