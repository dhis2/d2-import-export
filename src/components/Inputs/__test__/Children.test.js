import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Children } from '../Children'

jest.mock('react-final-form')

describe('Input component - Children', () => {
    it('should render correctly', () => {
        const file = shallow(<Children />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
