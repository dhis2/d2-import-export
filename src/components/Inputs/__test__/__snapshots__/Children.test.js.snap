// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Children should render correctly 1`] = `
<Field>
  <Label>
    Children
  </Label>
  <RadioGroup
    label="Children"
    name="children"
    options={
      Array [
        Object {
          "label": "Include descendants of organisation unit",
          "value": "true",
        },
        Object {
          "label": "Selected organisation unit",
          "value": "false",
        },
      ]
    }
  />
</Field>
`;
